custom/icds_reports/reports/take_home_ration.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import unicode_literals
+
+from corehq.apps.locations.models import SQLLocation
+from custom.icds_reports.utils import india_now, DATA_NOT_ENTERED
+from custom.icds_reports.models.views import TakeHomeRationMonthly
+
+
+class TakeHomeRationExport(object):
+    title = 'Take Home Ration'
+
+    def __init__(self, location, month, loc_level=0, beta=False):
+        self.location = location
+        self.loc_level = loc_level
+        self.month = month
+        self.beta = beta
+
+    def get_excel_data(self):
+
+        def _format_infrastructure_data(data):
+            return data if data is not None else DATA_NOT_ENTERED
+
+        filters = {"month": self.month, "aggregation_level": 5}
+
+        if self.loc_level == 4:
```

Comment:
This `loc_level` is quite common in the dashboard, so I didn't catch this earlier, but this seems unnecessary here. Instead of using some `loc_level`, you could check the location type's code here `self.location.type.name == 'supervisor'` or something similar. I think that might make it easier to read and may be a good change to these exports in general.

Reply:
Ohh... Sorry seeing it now. That's a good Idea. self.location is location_id though so might have to pull from SQLocation but overall sounds good to me. OK, Will be pushing a followup PR for this One. I think I can clean it up even further. 
Thanks,

Reply:
here: https://github.com/dimagi/commcare-hq/pull/24729